### PR TITLE
Shreyansh | Week5 - Implementation of Counter using Redux

### PR DESCRIPTION
**Pull request by shreyansh-khatri**
Pull Request Questionnaire

1) Briefly describe your change.
 Implemented Counter using redux and integrated it with main assignment.

2) What parts of the application are affected by your change?
My changes only affected the week5-Shreyansh branch and the components inside it.

3) How did you test your change?
I tested various parts such as validation, different inputs and some other edge cases.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -31,6 +32,7 @@ const App: React.FC = () => {
           </Route>
           <Route path='shreyansh' element={<Shreyansh />}>
             <Route path='week4/*' element={<ShreyanshWeek4 />}></Route>
+            <Route path='week5/*' element={<ShreyanshWeek5 />}></Route>
```
Comment (aditya-s-19) [inline src/App.tsx]: use self closing tag

File: src/components/Home/Shreyansh/Week5/Counter/CounterStyle.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    width: 100%;
+    height: 100%;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+  }
+  h1 {
+    font-size: 60px;
+    margin-bottom: 0px;
+    margin-top: 50px;
```
Comment (aditya-s-19) [inline src/components/Home/Shreyansh/Week5/Counter/CounterStyle.ts]: use a single margin for both these styles

File: src/components/Home/Shreyansh/Week5/Counter/Counter.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { CounterWrapper } from './CounterStyle';
+import { RootState } from './interface';
+
+const Counter: React.FC = () => {
+  const updateCounter: number = useSelector((state: RootState) => state.updateCounter);
+
+  const dispatch = useDispatch();
+  return (
+      <CounterWrapper>
+        <div className='mainContainer'>
```
Comment (choudhary-keshav) [inline src/components/Home/Shreyansh/Week5/Counter/Counter.tsx]: There is no need of extra div, line 12 & 13 can be of a single line

**Review by choudhary-keshav — CHANGES_REQUESTED** [review]

File: src/components/Home/Shreyansh/index.tsx [hunk 4]
```diff
@@ -15,6 +15,14 @@ const Shreyansh: React.FC = () => {
...
+          <button
+            className={`weekButton ${selectedWeek === 'week5' ? 'active' : ''}`}
+            value='week5'
+            onClick={(e) => setSelectedWeek(e.currentTarget.value)}>
+            Week-5
+          </button>
+        </Link>
       </div>
       <Outlet />
     </ShreyanshWrapper>
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Shreyansh/index.tsx]: It is been asked to remove this name from the css file.Dont use person's name in main component file

**Review by choudhary-keshav — DISMISSED** [review]

**Review by RitikaGarg-ITT — DISMISSED** [review]

**Review by aditya-s-19 — DISMISSED** [review]

**Review by shreegovinda — DISMISSED** [review]

**Review by choudhary-keshav — APPROVED** [review]

**Review by aditya-s-19 — APPROVED** [review]
